test/distributed/_composable/test_checkpoint.py

```diff
@@ -219,6 +221,112 @@ def forward(self, x):
 
         self.assertEqual(None, checkpoint.state(m)._ac_generator)
 
+    def test_checkpoint_kwargs(self):
+        class MyModel(torch.nn.Module):
+            def __init__(self, raise_exp: bool, change_shape_in_recomp: bool):
+                super().__init__()
+                self.fwd_count = 0
+                self.raise_exp = raise_exp
+                self.change_shape_in_recomp = change_shape_in_recomp
+                self.a = torch.nn.Linear(2, 2)
+
+            def forward(self, x):
+                if self.raise_exp and self.fwd_count == 0:
+                    raise RuntimeError("foo")
+                if self.raise_exp and self.fwd_count == 1:
+                    raise RuntimeError("bar")
+                if self.change_shape_in_recomp and self.fwd_count == 1:
+                    x.relu_()
+                random_tensor = torch.randn(1, 2)
+                x = self.a(x + random_tensor)
+                self.fwd_count += 1
+                return x
+
+        m = MyModel(True, False)
+        m0, m1, m2, m3 = (deepcopy(m) for _ in range(4))
+
+        # composable checkpoint does not support use_reentrant=True
+        with self.assertRaises(NotImplementedError):
+            checkpoint(m, use_reentrant=True)
+
+        # check giving an unsupported kwarg
+        with self.assertRaises(ValueError):
+            checkpoint(m0, foo="bar")
```

Comment:
If you have the extra motivation, you could use `assertRaisesRegex` to show that we are raising the right error message 😃

Reply:
I have the motivation 💪